static/js/unread.js

```diff
@@ -266,6 +267,14 @@ class UnreadTopicCounter {
             }
         }
 
+        for (const message_id of unread_mentions_counter) {
+            const bucket = this.bucketer.reverse_lookup.get(message_id);
+            const stream_id = bucket.bucket_id;
+            if (!res.streams_with_mentions.includes(stream_id)) {
+                res.streams_with_mentions.push(stream_id);
```

Comment:
Make this a `new Set`, and then you don't need to do this logic, which is quadratic.

Reply:
Updated the PR doing it. Thanks!